### PR TITLE
Refactor/chat sequence

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**

### CODE REVIEW

File: src/utils/openaiChat.ts [hunk 1]
```diff
@@ -107,128 +110,144 @@ async function SummaryGenerate(question: string, ans: string): Promise<string> {
...
-        : results.body;
-        return isQuestionOnStackExchange
-        ? null
-        : {
-          title: results.title,
-          snippet: snippet,
-          link: results.url,
-        };
-      });
+        source.type === "question" &&
```
Comment (urvishp80) [inline src/utils/openaiChat.ts:216]: Can we add one more processing for the mailing-list? In this case, if it is a mailing-list then we use a summary of it as the final text from this result else we use the body if the summary is not present.

**Review by urvishp80 — CHANGES_REQUESTED** [review]

File: src/pages/api/server.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// import { fetchESResult } from "@/utils/fetchESResult";
+import ERROR_MESSAGES from "@/config/error-config";
+import { processInput } from "@/utils/openaiChat";
+
+export const config: PageConfig = {
+  runtime: "edge",
+};
+
+export const fetchESResult = async (query: string, author?: string) : Promise<any[] | null> => {
+  const response = await fetch("http://localhost:3000/api/search", {
```
[the remark this answers is not in the document]
Comment (Emmanuel-Develops) [reply src/pages/api/server.ts]: It has been addressed in [3fc36d6](https://github.com/bitcoinsearch/chat-btc/pull/25/commits/3fc36d67d01e0b8ff56fe81af2508db91db36800) 

yeah, it was during testing. th eurl wouldn't work in deployment as that exists only locally

File: src/utils/openaiChat.ts (hunk 1 above)
Comment (Emmanuel-Develops) [reply src/utils/openaiChat.ts:216]: Oh, okay. After the markdown check?

Comment (urvishp80) [reply src/utils/openaiChat.ts:216]: yes

**Comment by adamjonas** [comment]
In a side by side comparison comparison I'm getting more answers from prod from this branch. How can we verify?

**Comment by Emmanuel-Develops** [comment]
> In a side by side comparison comparison I'm getting more answers from prod from this branch. How can we verify?

Yeah I noticed it too. We keep hitting the max token limit from open ai and that returns a 403 error.
So we return a fallback error message

Screenshot 2023-08-12 at 3 38 26 PM

**Comment by adamjonas** [comment]
You are saying we are hitting this limit just in the streaming branch?

**Comment by adamjonas** [comment]
Still not performing as well:
Screenshot 2023-08-18 at 4 04 53 AM
Screenshot 2023-08-18 at 4 05 04 AM

**Comment by Emmanuel-Develops** [comment]
@adamjonas after extensive debugging there are no differences between the final prompt text passed to chat on this branch and that on prod. Retrying the same prompt on prod locally failed to produce any results and prod on live didn't as well.
I used the same base query for all tests: `what are the most important functions in bitcoin core`, and disabled caching and reading cached answers so I only get chat responses.

I'll chalk it up to chatgpt not being exactly reproducible.

This branch did give an answer in one of my tests but that was 1 in about 8 instances of the same query so that isn't really usable.
Screenshot 2023-08-29 at 10 29 53 AM
Screenshot 2023-08-29 at 11 31 08 AM
Screenshot 2023-08-29 at 12 34 29 PM

IMG_01: one time answer from this branch
IMG_02: prod branch on local failing the same query
IMG_03: prod on live failing a reworded query (if not reworded it uses the cached no answer found message)

cc: @Extheoisah @urvishp80

**Comment by adamjonas** [comment]
Payments aren't working

**Comment by adamjonas** [comment]
This isn't a blocker to merge but i'd like to unify the payment options rather than have 2 tabs:

Screenshot 2023-09-10 at 8 10 56 AM
Screenshot 2023-09-10 at 8 10 51 AM

We should present 3 options and just drop the 6-hour one

**Pull request merged**